app/code/Magento/Customer/Api/Data/AttributeMetadataInterface.php

```diff
@@ -338,4 +340,34 @@ public function setIsFilterableInGrid($isFilterableInGrid);
      * @return $this
      */
     public function setIsSearchableInGrid($isSearchableInGrid);
+
+    /**
+     * Get attribute id
+     *
+     * @return int
+     */
+    public function getAttributeId();
```

Comment:
we can't modify this interface, it's public api
there's also no extension attributes so we would use them
we have to think of another approach
If it can't be done another way, then we might have to cut scope